examples/object-closure/object_closure.md

```diff
@@ -0,0 +1,12 @@
+# Object closure
+
+The object constructor and function can work as a closure, which captures its surrounding environment and can access variables outside of its own scope. This means that a closure can access variables that are defined in a different scope, such as the global scope or the scope of a parent.
```

Comment:
"surrounding environment" is a bit vague

Reply:
Updated 3dc00e058a6b8cd72dd07c254cdf9295db49bd89